metacatalog_api/router/api/read.py

```
@@ -83,10 +85,53 @@ def get_author_by_name(id: int = None, name: str = None, search: str = None):
-        return HTTPException(status_code=404, detail=str(e))
+        raise HTTPException(status_code=404, detail=str(e)) from e
 
     return variables
+
+@read_router.get('/variable/{id}')
+@read_router.get('/variable/{id}.json')
+def get_variable(id: int):
+    try:
+        variable = core.variables(id=id)
+    except Exception as e:
+        raise HTTPException(status_code=404, detail=str(e))
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)